### PR TITLE
fix: use merkle tree hook address from registry for self relay hook derivations

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Circumvents bad hook derivation during self relay

### Related issues

- Fixes https://github.com/hyperlane-xyz/issues/issues/1320

### Backward compatibility

Yes

### Testing

manual warp send from sepolia origin

### CODE REVIEW

File: typescript/cli/src/utils/relay.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { HookType, HyperlaneRelayer } from '@hyperlane-xyz/sdk';
+
+// hack for bypassing bad hook derivation when self-relaying
```
Comment (nbayindirli) [inline typescript/cli/src/utils/relay.ts:3]: nit
```suggestion
/**
* Workaround helper for bypassing bad hook derivation when self-relaying.
*/
```

**Review by nbayindirli — APPROVED** [review]
local tests successful 🚢

File: typescript/cli/src/utils/relay.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export function stubMerkleTreeConfig(
+  relayer: HyperlaneRelayer,
+  chain: string,
+  hookAddress: string,
+  merkleAddress: string,
+) {
+  relayer.hydrate({
```
[the remark this answers is not in the document]
Comment (yorhodes) [reply typescript/cli/src/utils/relay.ts:10]: it is not already hydrated so not sure why this makes more sense
given that I didnt introduce this function on this PR probably will skip

[the remark this answers is not in the document]
Comment (yorhodes) [reply typescript/cli/src/utils/relay.ts:12]: honestly not sure whether it should be called "hydration" in the first place but I think this is apt
https://www.baeldung.com/java-object-hydration

**Pull request merged**